File: check_db.js
import sqlite3 from 'sqlite3';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const dbPath = path.resolve(__dirname, 'server/database.sqlite');

console.log(`Checking DB at: ${dbPath}`);
const db = new sqlite3.Database(dbPath, (err) => {
    if (err) {
        console.error('Error opening database', err.message);
        process.exit(1);
    }
    console.log('Connected.');
});

db.all('SELECT * FROM products LIMIT 1', [], (err, rows) => {
    if (err) {
        console.error('Error querying products:', err.message);
    } else {
        console.log('Products query success. Rows:', rows);
    }

    db.all("SELECT name FROM sqlite_master WHERE type='table';", [], (err, tables) => {
        if (err) console.error("Error listing tables:", err);
        else console.log("Tables:", tables);
    });
});

File: check_db.py
import sqlite3
import sys

db_path = r'C:\Users\Usuario\Desktop\desarollo\LBDC\server\database.sqlite'
try:
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()
    
    # Get all tables
    cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")
    tables = cursor.fetchall()
    print('Tablas en la base de datos:')
    for table in tables:
        print(f'  {table[0]}')
    
    # Check categories table
    print('\n=== Categorias ===')
    try:
        cursor.execute('SELECT * FROM categories')
        rows = cursor.fetchall()
        if rows:
            # Get column names
            cursor.execute('PRAGMA table_info(categories)')
            cols = cursor.fetchall()
            col_names = [col[1] for col in cols]
            print('Columnas:', col_names)
            print(f'Total categorías: {len(rows)}')
            for row in rows[:10]:
                print(row)
        else:
            print('No hay categorías')
    except Exception as e:
        print(f'Error: {e}')
    
    # Check products table structure
    print('\n=== Estructura de products ===')
    try:
        cursor.execute('PRAGMA table_info(products)')
        cols = cursor.fetchall()
        for col in cols:
            print(f'  {col[1]} ({col[2]})')
    except Exception as e:
        print(f'Error: {e}')
    
    # Count products
    print('\n=== Conteo de productos ===')
    try:
        cursor.execute('SELECT COUNT(*) FROM products')
        count = cursor.fetchone()[0]
        print(f'Total productos: {count}')
    except Exception as e:
        print(f'Error: {e}')
    
    conn.close()
except Exception as e:
    print(f'Error de conexión: {e}')